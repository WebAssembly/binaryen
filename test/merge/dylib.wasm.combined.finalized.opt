(module
 (type $1 (func (param i32) (result i32)))
 (type $2 (func (result i32)))
 (type $3 (func))
 (import "env" "_puts" (func $import$1 (param i32) (result i32)))
 (import "env" "memory" (memory $0 256))
 (import "env" "table" (table 8 anyfunc))
 (global $global$0 (mut i32) (i32.const 0))
 (global $global$1 (mut i32) (i32.const 0))
 (global $global$2 i32 (i32.const 0))
 (global $global$0$0 (mut i32) (i32.const 0))
 (global $global$1$0 (mut i32) (i32.const 0))
 
 (data (i32.const 1024) "hello, world!")
 (export "__post_instantiate" (func $__post_instantiate))
 (export "_main" (func $_main))
 (export "runPostSets" (func $runPostSets))
 (export "_str" (global $global$2))
 (export "_foo" (func $_foo))
 (func $_main (type $2) (result i32)
  (block $label$0 (result i32)
   (block $label$1 (result i32)
    (drop
     (call $import$1
      (i32.const 1024)
     )
    )
    (i32.const 0)
   )
  )
 )
 (func $runPostSets (type $3)
  (nop)
 )
 (func $__post_instantiate (type $3)
  (call $__post_instantiate$0)
  (block
   (block $label$0
    (block $label$1
     (set_global $global$0
      (i32.const 1040)
     )
     (set_global $global$1
      (i32.add
       (get_global $global$0)
       (i32.const 32)
      )
     )
     (call $runPostSets)
    )
   )
  )
 )
 (func $_foo (type $2) (result i32)
  (local $var$0 i32)
  (block $label$0 (result i32)
   (block $label$1 (result i32)
    (set_local $var$0
     (call $_main)
    )
    (get_local $var$0)
   )
  )
 )
 (func $runPostSets$0 (type $3)
  (nop)
 )
 (func $__post_instantiate$0 (type $3)
  (block $label$0
   (block $label$1
    (set_global $global$0$0
     (i32.const 1072)
    )
    (set_global $global$1$0
     (i32.add
      (get_global $global$0$0)
      (i32.const 10)
     )
    )
    (call $runPostSets$0)
   )
  )
 )
 ;; custom section "dylink", size 2
)
