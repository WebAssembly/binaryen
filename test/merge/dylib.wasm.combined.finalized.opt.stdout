merged total memory size: 64
merged total table size: 0
merged functions: 6
